# Analyzing Used Car Listings on eBay Kleinanzeigen
**We will be working on a dataset of used cars from eBay Kleinanzeigen, a classifieds section of the German eBay website.**

The dataset was originally scraped and uploaded to Kaggle. The version of the dataset we are working with is a sample of 50,000 data points that was prepared by Dataquest including simulating a less-cleaned version of the data.

The data dictionary provided with data is as follows:

- dateCrawled - When this ad was first crawled. All field-values are taken from this date.
- name - Name of the car.
- seller - Whether the seller is private or a dealer.
- offerType - The type of listing
- price - The price on the ad to sell the car.
- abtest - Whether the listing is included in an A/B test.
- vehicleType - The vehicle Type.
- yearOfRegistration - The year in which which year the car was first registered.
- gearbox - The transmission type.
- powerPS - The power of the car in PS.
- model - The car model name.
- kilometer - How many kilometers the car has driven.
- monthOfRegistration - The month in which which year the car was first registered.
- fuelType - What type of fuel the car uses.
- brand - The brand of the car.
- notRepairedDamage - If the car has a damage which is not yet repaired.
- dateCreated - The date on which the eBay listing was created.
- nrOfPictures - The number of pictures in the ad.
- postalCode - The postal code for the location of the vehicle.
- lastSeenOnline - When the crawler saw this ad last online.

The aim of this project is to clean the data and analyze the included used car listings.



In [1]:
import numpy as np
import pandas as pd

In [2]:
autos = pd.read_csv('/Users/number1/2024/Data/Project 3/autos.csv',delimiter=',',encoding='Latin-1')

In [3]:
autos.info()
autos.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 371528 entries, 0 to 371527
Data columns (total 20 columns):
 #   Column               Non-Null Count   Dtype 
---  ------               --------------   ----- 
 0   dateCrawled          371528 non-null  object
 1   name                 371528 non-null  object
 2   seller               371528 non-null  object
 3   offerType            371528 non-null  object
 4   price                371528 non-null  int64 
 5   abtest               371528 non-null  object
 6   vehicleType          333659 non-null  object
 7   yearOfRegistration   371528 non-null  int64 
 8   gearbox              351319 non-null  object
 9   powerPS              371528 non-null  int64 
 10  model                351044 non-null  object
 11  kilometer            371528 non-null  int64 
 12  monthOfRegistration  371528 non-null  int64 
 13  fuelType             338142 non-null  object
 14  brand                371528 non-null  object
 15  notRepairedDamage    299468 non-nu

dateCrawled                            name  seller offerType  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat   Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat   Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat   Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat   Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat   Angebot   

   price abtest vehicleType  yearOfRegistration    gearbox  powerPS  model  \
0    480   test         NaN                1993    manuell        0   golf   
1  18300   test       coupe                2011    manuell      190    NaN   
2   9800   test         suv                2004  automatik      163  grand   
3   1500   test  kleinwagen                2001    manuell       75   golf   
4   3600   test  kleinwagen                2008    manuell       69  fabia   

   kilometer  monthOfRegistration fuelType       brand notRepairedDamage  \
0     150000                    0   benzin  volkswagen               NaN   
1     125000                    5   diesel        audi                ja   
2     125000                    8   diesel        jeep               NaN   
3     150000                    6   benzin  volkswagen              nein   
4      90000                    7   diesel       skoda              nein   

           dateCreated  nrOfPictures  postalCode             lastSeen  
0  2016-03-24 00:00:00             0       70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00             0       66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00             0       90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00             0       91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00             0       60437  2016-04-06 10:17:21

Column's names aren't easy to use. Some columns like: vehicle_type, gearbox, model, fuel type and notRepairedDamage have some NaN values

We'll make a few changes here:

- Change the columns from camelcase to snakecase.
- Change a few wordings to more accurately describe the columns.

In [4]:
#Check columns's name:
autos.columns = ['date_crawled','name','seller','offer_type','price','ab_test', 'vehicle_type','registration_year',\
                 'gearbox', 'power_ps','model','odometer','registration_month','fuel_type','brand','unrepaired_damgage',\
                 'ad_created','num_photos','postal_code', 'last_seen']

In [5]:
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'ab_test',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damgage', 'ad_created', 'num_photos', 'postal_code',
       'last_seen'],
      dtype='object')

In [6]:
autos.head()

date_crawled                            name  seller offer_type  \
0  2016-03-24 11:52:17                      Golf_3_1.6  privat    Angebot   
1  2016-03-24 10:58:45            A5_Sportback_2.7_Tdi  privat    Angebot   
2  2016-03-14 12:52:21  Jeep_Grand_Cherokee_"Overland"  privat    Angebot   
3  2016-03-17 16:54:04              GOLF_4_1_4__3TÜRER  privat    Angebot   
4  2016-03-31 17:25:20  Skoda_Fabia_1.4_TDI_PD_Classic  privat    Angebot   

   price ab_test vehicle_type  registration_year    gearbox  power_ps  model  \
0    480    test          NaN               1993    manuell         0   golf   
1  18300    test        coupe               2011    manuell       190    NaN   
2   9800    test          suv               2004  automatik       163  grand   
3   1500    test   kleinwagen               2001    manuell        75   golf   
4   3600    test   kleinwagen               2008    manuell        69  fabia   

   odometer  registration_month fuel_type       brand unrepaired_damgage  \
0    150000                   0    benzin  volkswagen                NaN   
1    125000                   5    diesel        audi                 ja   
2    125000                   8    diesel        jeep                NaN   
3    150000                   6    benzin  volkswagen               nein   
4     90000                   7    diesel       skoda               nein   

            ad_created  num_photos  postal_code            last_seen  
0  2016-03-24 00:00:00           0        70435  2016-04-07 03:16:57  
1  2016-03-24 00:00:00           0        66954  2016-04-07 01:46:50  
2  2016-03-14 00:00:00           0        90480  2016-04-05 12:47:46  
3  2016-03-17 00:00:00           0        91074  2016-03-17 17:40:17  
4  2016-03-31 00:00:00           0        60437  2016-04-06 10:17:21

In [7]:
autos.describe(include = 'all')

date_crawled         name  seller offer_type         price  \
count                371528       371528  371528     371528  3.715280e+05   
unique               280500       233531       2          2           NaN   
top     2016-03-24 14:49:47  Ford_Fiesta  privat    Angebot           NaN   
freq                      7          657  371525     371516           NaN   
mean                    NaN          NaN     NaN        NaN  1.729514e+04   
std                     NaN          NaN     NaN        NaN  3.587954e+06   
min                     NaN          NaN     NaN        NaN  0.000000e+00   
25%                     NaN          NaN     NaN        NaN  1.150000e+03   
50%                     NaN          NaN     NaN        NaN  2.950000e+03   
75%                     NaN          NaN     NaN        NaN  7.200000e+03   
max                     NaN          NaN     NaN        NaN  2.147484e+09   

       ab_test vehicle_type  registration_year  gearbox       power_ps  \
count   371528       333659      371528.000000   351319  371528.000000   
unique       2            8                NaN        2            NaN   
top       test    limousine                NaN  manuell            NaN   
freq    192585        95894                NaN   274214            NaN   
mean       NaN          NaN        2004.577997      NaN     115.549477   
std        NaN          NaN          92.866598      NaN     192.139578   
min        NaN          NaN        1000.000000      NaN       0.000000   
25%        NaN          NaN        1999.000000      NaN      70.000000   
50%        NaN          NaN        2003.000000      NaN     105.000000   
75%        NaN          NaN        2008.000000      NaN     150.000000   
max        NaN          NaN        9999.000000      NaN   20000.000000   

         model       odometer  registration_month fuel_type       brand  \
count   351044  371528.000000       371528.000000    338142      371528   
unique     251            NaN                 NaN         7          40   
top       golf            NaN                 NaN    benzin  volkswagen   
freq     30070            NaN                 NaN    223857       79640   
mean       NaN  125618.688228            5.734445       NaN         NaN   
std        NaN   40112.337051            3.712412       NaN         NaN   
min        NaN    5000.000000            0.000000       NaN         NaN   
25%        NaN  125000.000000            3.000000       NaN         NaN   
50%        NaN  150000.000000            6.000000       NaN         NaN   
75%        NaN  150000.000000            9.000000       NaN         NaN   
max        NaN  150000.000000           12.000000       NaN         NaN   

       unrepaired_damgage           ad_created  num_photos   postal_code  \
count              299468               371528    371528.0  371528.00000   
unique                  2                  114         NaN           NaN   
top                  nein  2016-04-03 00:00:00         NaN           NaN   
freq               263182                14450         NaN           NaN   
mean                  NaN                  NaN         0.0   50820.66764   
std                   NaN                  NaN         0.0   25799.08247   
min                   NaN                  NaN         0.0    1067.00000   
25%                   NaN                  NaN         0.0   30459.00000   
50%                   NaN                  NaN         0.0   49610.00000   
75%                   NaN                  NaN         0.0   71546.00000   
max                   NaN                  NaN         0.0   99998.00000   

                  last_seen  
count                371528  
unique               182806  
top     2016-04-07 06:45:59  
freq                     17  
mean                    NaN  
std                     NaN  
min                     NaN  
25%                     NaN  
50%                     NaN  
75%                     NaN  
max                     NaN

seller column is mispelled ( 'privat').

vehicle_type have NaN values.

gearbox have NaN values.

power_ps = 0

model have NaN values.

fuel_type have NaN values.

registration_month = 0

all value in num_photos are 0

In [8]:
autos['seller'].value_counts()

privat        371525
gewerblich         3
Name: seller, dtype: int64

In [9]:
#Fix privat to private
autos['seller'] = autos['seller'].map({'privat':'private','gewerblich':'gewerblich'})

In [10]:
autos['seller'].head()

0    private
1    private
2    private
3    private
4    private
Name: seller, dtype: object

In [11]:
#num_photos column:
autos['num_photos'].value_counts()
#All value in num_photos = 0. Let's delete it
autos = autos.drop('num_photos',axis=1)
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'ab_test',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damgage', 'ad_created', 'postal_code', 'last_seen'],
      dtype='object')

In [12]:
autos.columns

Index(['date_crawled', 'name', 'seller', 'offer_type', 'price', 'ab_test',
       'vehicle_type', 'registration_year', 'gearbox', 'power_ps', 'model',
       'odometer', 'registration_month', 'fuel_type', 'brand',
       'unrepaired_damgage', 'ad_created', 'postal_code', 'last_seen'],
      dtype='object')

In [13]:
autos['odometer'] = autos['odometer'].astype(str) + 'km'
autos['odometer'].head()


0    150000km
1    125000km
2    125000km
3    150000km
4     90000km
Name: odometer, dtype: object

In [14]:
#Fix odometer column and change it to int type
autos['odometer'] = autos['odometer'].str.replace('km','').astype(int)
autos['odometer'].dtype

dtype('int64')

In [15]:
autos['odometer'].head()

0    150000
1    125000
2    125000
3    150000
4     90000
Name: odometer, dtype: int64

In [16]:
#rename it to odometer_km
autos.rename({'odometer':'odometer_km'},axis=1,inplace=True)
autos['odometer_km']

0         150000
1         125000
2         125000
3         150000
4          90000
           ...  
371523     20000
371524    125000
371525    150000
371526    150000
371527     50000
Name: odometer_km, Length: 371528, dtype: int64

In [17]:
autos['odometer_km'].value_counts()

150000    240797
125000     38067
100000     15920
90000      12523
80000      11053
70000       9773
60000       8669
50000       7615
5000        7069
40000       6376
30000       6041
20000       5676
10000       1949
Name: odometer_km, dtype: int64

In [18]:
autos['odometer_km'].describe()

count    371528.000000
mean     125618.688228
std       40112.337051
min        5000.000000
25%      125000.000000
50%      150000.000000
75%      150000.000000
max      150000.000000
Name: odometer_km, dtype: float64

We can see that the values in this field are rounded, which might indicate that sellers had to choose from pre-set options for this field. Additionally, there are more high mileage than low mileage vehicles.

In [19]:
print(autos['price'].unique().shape)
print('\n')
print(autos['price'].describe())
print('\n')
autos['price'].value_counts().head(25)


(5597,)


count    3.715280e+05
mean     1.729514e+04
std      3.587954e+06
min      0.000000e+00
25%      1.150000e+03
50%      2.950000e+03
75%      7.200000e+03
max      2.147484e+09
Name: price, dtype: float64




0       10778
500      5670
1500     5394
1000     4649
1200     4594
2500     4438
600      3819
3500     3792
800      3784
2000     3432
999      3364
750      3203
650      3150
4500     3053
850      2946
2200     2936
700      2936
1800     2886
900      2874
950      2793
1100     2772
1300     2757
300      2731
3000     2720
550      2591
Name: price, dtype: int64

Again, the prices in this column seem rounded, however given there are 5597 unique values in the column, that may just be people's tendency to round prices on the site.

There are 10778 cars listed with $0 price - given that this is only 3% of the of the cars, we might consider removing these rows. The maximum price is 2,14 billion dollars, which seems a lot, let's look at the highest prices further.

In [20]:
autos["price"].value_counts().sort_index(ascending=False).head(23)

2147483647     1
99999999      15
99000000       1
74185296       1
32545461       1
27322222       1
14000500       1
12345678       9
11111111      10
10010011       1
10000000       8
9999999        3
3895000        1
3890000        1
2995000        1
2795000        1
1600000        2
1300000        1
1250000        2
1234566        1
1111111        2
1010010        1
1000000        5
Name: price, dtype: int64

In [21]:
autos["price"].value_counts().sort_index(ascending=False).head(22).sum()

65

In [22]:
autos['price'].value_counts().sort_index(ascending=True).head(20)

0     10778
1      1189
2        12
3         8
4         1
5        26
7         3
8         9
9         8
10       84
11        5
12        8
13        7
14        5
15       27
16        2
17        5
18        3
19        3
20       51
Name: price, dtype: int64

There are a number of listings with prices below 0. There are also a small number of listings with very high values, including 65 over $1 million.

Given that eBay is an auction site, there could legitimately be items where the opening bid is 1 items, but remove anything above 1 million, since it seems that prices increase steadily to that number and then jump up to less realistic numbers.

In [23]:
autos = autos[autos['price'].between(1,1000000)]
autos['price'].describe()

count     360685.000000
mean        5999.191366
std        12634.935399
min            1.000000
25%         1250.000000
50%         3000.000000
75%         7490.000000
max      1000000.000000
Name: price, dtype: float64

Look like in this web people usually sell old car for low price. Which have the average odometer at about 125,000 km and average at 6,000$ for price.


## Exploring the date columns
There are a number of columns with date information:

- date_crawled
- registration_month
- registration_year
- ad_created
- last_seen

These are a combination of dates that were crawled, and dates with meta-information from the crawler. The non-registration dates are stored as strings.

We'll explore each of these columns to learn more about the listings.

In [24]:
autos[['date_crawled','registration_month','registration_year','ad_created','last_seen']].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 360685 entries, 0 to 371527
Data columns (total 5 columns):
 #   Column              Non-Null Count   Dtype 
---  ------              --------------   ----- 
 0   date_crawled        360685 non-null  object
 1   registration_month  360685 non-null  int64 
 2   registration_year   360685 non-null  int64 
 3   ad_created          360685 non-null  object
 4   last_seen           360685 non-null  object
dtypes: int64(2), object(3)
memory usage: 16.5+ MB


Right now, the date_crawled, last_seen, and ad_created columns are all identified as string values by pandas. Because these three columns are represented as strings, we need to convert the data into a numerical representation so we can understand it quantitatively. The other two columns are represented as numeric values.


Let's first understand how the values in the three string columns are formatted. These columns all represent full timestamp values:

In [25]:
autos[['date_crawled','ad_created','last_seen']].head(5)

date_crawled           ad_created            last_seen
0  2016-03-24 11:52:17  2016-03-24 00:00:00  2016-04-07 03:16:57
1  2016-03-24 10:58:45  2016-03-24 00:00:00  2016-04-07 01:46:50
2  2016-03-14 12:52:21  2016-03-14 00:00:00  2016-04-05 12:47:46
3  2016-03-17 16:54:04  2016-03-17 00:00:00  2016-03-17 17:40:17
4  2016-03-31 17:25:20  2016-03-31 00:00:00  2016-04-06 10:17:21

You'll notice that the first 10 characters represent the day (e.g. 2016-03-12). To understand the date range, we can extract just the date values, use Series.value_counts() to generate a distribution, and then sort by the index.

In [26]:
autos['date_crawled'].str[0:10].value_counts(dropna=False).sort_index()

2016-03-05     9214
2016-03-06     5223
2016-03-07    12859
2016-03-08    12071
2016-03-09    12304
2016-03-10    11775
2016-03-11    11820
2016-03-12    13072
2016-03-13     5692
2016-03-14    13103
2016-03-15    12056
2016-03-16    10893
2016-03-17    11418
2016-03-18     4731
2016-03-19    12721
2016-03-20    13129
2016-03-21    12874
2016-03-22    11719
2016-03-23    11543
2016-03-24    10789
2016-03-25    11829
2016-03-26    11531
2016-03-27    10904
2016-03-28    12646
2016-03-29    12308
2016-03-30    12094
2016-03-31    11497
2016-04-01    12318
2016-04-02    12658
2016-04-03    13999
2016-04-04    13571
2016-04-05     4612
2016-04-06     1128
2016-04-07      584
Name: date_crawled, dtype: int64

In [27]:
autos['date_crawled'].str[0:10].value_counts(dropna=False).sort_values()

2016-04-07      584
2016-04-06     1128
2016-04-05     4612
2016-03-18     4731
2016-03-06     5223
2016-03-13     5692
2016-03-05     9214
2016-03-24    10789
2016-03-16    10893
2016-03-27    10904
2016-03-17    11418
2016-03-31    11497
2016-03-26    11531
2016-03-23    11543
2016-03-22    11719
2016-03-10    11775
2016-03-11    11820
2016-03-25    11829
2016-03-15    12056
2016-03-08    12071
2016-03-30    12094
2016-03-09    12304
2016-03-29    12308
2016-04-01    12318
2016-03-28    12646
2016-04-02    12658
2016-03-19    12721
2016-03-07    12859
2016-03-21    12874
2016-03-12    13072
2016-03-14    13103
2016-03-20    13129
2016-04-04    13571
2016-04-03    13999
Name: date_crawled, dtype: int64

Looks like the site was crawled daily over roughly a one month period in March and April 2016. The distribution of listings crawled on each day is roughly uniform.

In [28]:
autos['last_seen'].str[:10].value_counts(dropna=False).sort_index()

2016-03-05      456
2016-03-06     1478
2016-03-07     1876
2016-03-08     2863
2016-03-09     3543
2016-03-10     4133
2016-03-11     4672
2016-03-12     8383
2016-03-13     3033
2016-03-14     4391
2016-03-15     5888
2016-03-16     5921
2016-03-17    10353
2016-03-18     2485
2016-03-19     5891
2016-03-20     7173
2016-03-21     7224
2016-03-22     7397
2016-03-23     6499
2016-03-24     6911
2016-03-25     6852
2016-03-26     5755
2016-03-27     6031
2016-03-28     8004
2016-03-29     8399
2016-03-30     8556
2016-03-31     8743
2016-04-01     8620
2016-04-02     9006
2016-04-03     9132
2016-04-04     9209
2016-04-05    45795
2016-04-06    78970
2016-04-07    47043
Name: last_seen, dtype: int64

The last three days contain a disproportionate amount of 'last seen' values. Given that these are 5-9x the values from the previous days, it's unlikely that there was a massive spike in sales, and more likely that these values are to do with the crawling period ending and don't indicate car sales.

In [29]:
autos['ad_created'].str[:10].value_counts(dropna=False).sort_index()

2014-03-10        1
2015-03-20        1
2015-06-11        1
2015-06-18        1
2015-08-07        1
              ...  
2016-04-03    14067
2016-04-04    13610
2016-04-05     4191
2016-04-06     1125
2016-04-07      561
Name: ad_created, Length: 114, dtype: int64

There is a large variety of ad created dates. Most fall within 1-2 months of the listing date, but a few are quite old, with the oldest at around 2 years.

## Dealing with Incorrect Registration Year Data

In [30]:
autos['registration_year'].describe()

count    360685.000000
mean       2004.451821
std          82.111504
min        1000.000000
25%        1999.000000
50%        2004.000000
75%        2008.000000
max        9999.000000
Name: registration_year, dtype: float64

The year that the car was first registered will likely indicate the age of the car. Looking at this column, we note some odd values. The minimum value is 1000, long before cars were invented and the maximum is 9999, many years into the future.

Because a car can't be first registered before the listing was seen, any vehicle with a registration year above 2016 is definitely inaccurate. Determining the earliest valid year is more difficult. Realistically, it could be somewhere in the first few decades of the 1900s.

One option is to remove the listings with these values. Let's determine what percentage of our data has invalid values in this column:

In [31]:
print(((~autos['registration_year'].between(1900,2016)).sum()) / autos.shape[0])

0.0387512649541844


Given that this is less than 4% of our data, we will remove these rows.

In [32]:
autos = autos[autos['registration_year'].between(1900,2016)]

In [33]:
autos['registration_year'].value_counts(dropna=False).sort_index().tail(50)

1967      136
1968      144
1969      153
1970      321
1971      191
1972      245
1973      196
1974      198
1975      146
1976      175
1977      189
1978      298
1979      296
1980      531
1981      270
1982      319
1983      427
1984      405
1985      696
1986      540
1987      657
1988      926
1989     1259
1990     2503
1991     2657
1992     2897
1993     3299
1994     4646
1995     8999
1996    10410
1997    14030
1998    17229
1999    22031
2000    23126
2001    19745
2002    18820
2003    19606
2004    19475
2005    21726
2006    20005
2007    17506
2008    16035
2009    15499
2010    12271
2011    11996
2012     9359
2013     6122
2014     4769
2015     2923
2016     9224
Name: registration_year, dtype: int64

It appears that most of the vehicles were first registered in the past 20 years.

## Exploring Price by Brand

In [34]:
autos['brand'].value_counts(normalize = True)

volkswagen        0.211682
bmw               0.109871
opel              0.106398
mercedes_benz     0.096837
audi              0.089539
ford              0.068914
renault           0.047510
peugeot           0.030149
fiat              0.025687
seat              0.018658
skoda             0.015685
mazda             0.015382
smart             0.014329
citroen           0.013948
nissan            0.013596
toyota            0.012930
hyundai           0.009971
sonstige_autos    0.009524
mini              0.009383
volvo             0.009146
mitsubishi        0.008235
honda             0.007531
kia               0.006914
suzuki            0.006366
alfa_romeo        0.006308
porsche           0.006253
chevrolet         0.005024
chrysler          0.003862
dacia             0.002495
jeep              0.002195
land_rover        0.002166
daihatsu          0.002160
subaru            0.002117
jaguar            0.001736
saab              0.001465
daewoo            0.001457
trabant           0.001408
l

In [35]:
autos['brand'].value_counts(normalize = True).head(5).sum()

0.6143267533486392

German manufacturers represent four out of the top five brands, almost 50% of the overall listings. Volkswagen is by far the most popular brand, with approximately double the cars for sale of the next two brands combined.

There are lots of brands that don't have a significant percentage of listings, so we will limit our analysis to brands representing more than 5% of total listings.

In [36]:

count_brand = autos['brand'].value_counts(normalize = True)


In [37]:
common_brand = count_brand[count_brand > 0.05].index

In [38]:
print(common_brand)

Index(['volkswagen', 'bmw', 'opel', 'mercedes_benz', 'audi', 'ford'], dtype='object')


In [39]:
mean_price_brand ={}
for c in common_brand:
    mean = autos[autos['brand']== c]['price'].mean()
    mean_price_brand[c] = int(mean)
print(mean_price_brand)

{'volkswagen': 5452, 'bmw': 8541, 'opel': 2998, 'mercedes_benz': 8609, 'audi': 9166, 'ford': 3779}


Of the top 5 brands, there is a distinct price gap:

Audi, BMW and Mercedes Benz are more expensive
Ford and Opel are less expensive
Volkswagen is in between - this may explain its popularity, it may be a 'best of 'both worlds' option.

## Exploring Mileage

In [40]:
mean_mileage_brand = {}
for c in common_brand:
    mean = autos[autos['brand']== c]['odometer_km'].mean()
    mean_mileage_brand[c]= int(mean)
print(mean_mileage_brand)


{'volkswagen': 128387, 'bmw': 132791, 'opel': 128720, 'mercedes_benz': 130566, 'audi': 129440, 'ford': 123663}


In [41]:
price = pd.Series(mean_price_brand)

In [42]:
print(price)

volkswagen       5452
bmw              8541
opel             2998
mercedes_benz    8609
audi             9166
ford             3779
dtype: int64


In [43]:
mileage = pd.Series(mean_mileage_brand)

In [44]:
brand_info = pd.DataFrame()

In [45]:
brand_info['mean_mileage'] = mileage

In [46]:
brand_info['mean_price'] = price

In [47]:
brand_info.head()

mean_mileage  mean_price
volkswagen           128387        5452
bmw                  132791        8541
opel                 128720        2998
mercedes_benz        130566        8609
audi                 129440        9166

The range of car mileages does not vary as much as the prices do by brand, instead all falling within 10% for the top brands. There is a slight trend to the more expensive vehicles having higher mileage, with the less expensive vehicles having lower mileage.